File: EzPay/src/main/java/com/thinhtran/EzPay/service/NotificationService.java
package com.thinhtran.EzPay.service;

import com.thinhtran.EzPay.entity.Notification;
import com.thinhtran.EzPay.entity.NotificationType;

import java.util.List;

public interface NotificationService {
    
    /**
     * Create and send notification
     */
    void createAndSendNotification(Long userId, NotificationType type, String title, String message, Long relatedId);
    
    /**
     * Get all notifications for a user
     */
    List<Notification> getUserNotifications(Long userId);
    
    /**
     * Get unread notifications for a user
     */
    List<Notification> getUnreadNotifications(Long userId);
    
    /**
     * Mark notification as read
     */
    void markAsRead(Long notificationId, Long userId);
    
    /**
     * Mark all notifications as read for a user
     */
    void markAllAsRead(Long userId);
    
    /**
     * Get unread count for a user
     */
    Long getUnreadCount(Long userId);
    
    /**
     * Get recent notifications (last 10)
     */
    List<Notification> getRecentNotifications(Long userId);
} 

File: frontend/src/services/notificationService.ts
import api from './api';
import { Notification } from './websocketService';

export interface ApiResponse<T> {
  code: string;
  message: string;
  data?: T;
}

export const notificationService = {
  // Get all notifications for current user
  getUserNotifications: async (): Promise<Notification[]> => {
    const response = await api.get<ApiResponse<Notification[]>>('/notifications');
    return response.data.data || [];
  },

  // Get unread notifications
  getUnreadNotifications: async (): Promise<Notification[]> => {
    const response = await api.get<ApiResponse<Notification[]>>('/notifications/unread');
    return response.data.data || [];
  },

  // Get recent notifications (last 10)
  getRecentNotifications: async (): Promise<Notification[]> => {
    const response = await api.get<ApiResponse<Notification[]>>('/notifications/recent');
    return response.data.data || [];
  },

  // Get unread count
  getUnreadCount: async (): Promise<number> => {
    const response = await api.get<ApiResponse<number>>('/notifications/unread-count');
    return response.data.data || 0;
  },

  // Mark notification as read
  markAsRead: async (notificationId: number): Promise<void> => {
    await api.put(`/notifications/${notificationId}/read`);
  },

  // Mark all notifications as read
  markAllAsRead: async (): Promise<void> => {
    await api.put('/notifications/read-all');
  },
}; 